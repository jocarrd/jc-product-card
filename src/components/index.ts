import { ProductCardHOCProps } from "../interfaces/interfaces";
import { ProductButtons } from "./ProductButtons";
import { PoductCard as ProductCardHOC } from "./ProductCard";
import { ProductImage } from "./ProductImage";
import { ProductTitle } from "./ProductTitle";

export{ ProductButtons } from "./ProductButtons";
export { ProductTitle } from "./ProductTitle";
export { ProductImage } from "./ProductImage";


export const ProductCard : ProductCardHOCProps = Object.assign(ProductCardHOC,{
    Title:ProductTitle,
    Image : ProductImage,
    Buttons: ProductButtons
})

