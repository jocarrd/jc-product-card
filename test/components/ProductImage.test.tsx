import React from 'react';
import renderer from 'react-test-renderer';
import { ProductImage, ProductCard } from '../../src/components';
import { product2 } from '../data/products';

const IMG_URL =
  'https://www.google.com/url?sa=i&url=https%3A%2F%2Fwww.rtve.es%2Fplay%2Faudios%2Fespacio-en-blanco%2Fovnis-primeras-revelaciones-del-pentagono-13-06-21%2F5940378%2F&psig=AOvVaw0p5oysTmPrZu8TAJEbq0B6&ust=1668962740310000&source=images&cd=vfe&ved=0CA0QjRxqFwoTCPjXo-zYuvsCFQAAAAAdAAAAABAD';

describe('ProductImage', () => {
  test('Product Image must be render with custom image', () => {
    const component = renderer.create(<ProductImage img={IMG_URL} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('ProductCard must show product image', () => {
    const component = renderer.create(
      <ProductCard product={product2}>{() => <ProductImage />}</ProductCard>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
