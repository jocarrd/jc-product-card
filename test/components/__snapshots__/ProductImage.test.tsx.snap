// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage Product Image must be render with custom image 1`] = `
<img
  alt="Product"
  className="productImg undefined"
  src="https://www.google.com/url?sa=i&url=https%3A%2F%2Fwww.rtve.es%2Fplay%2Faudios%2Fespacio-en-blanco%2Fovnis-primeras-revelaciones-del-pentagono-13-06-21%2F5940378%2F&psig=AOvVaw0p5oysTmPrZu8TAJEbq0B6&ust=1668962740310000&source=images&cd=vfe&ved=0CA0QjRxqFwoTCPjXo-zYuvsCFQAAAAAdAAAAABAD"
/>
`;

exports[`ProductImage ProductCard must show product image 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;
